### PR TITLE
Don't use magic

### PR DESCRIPTION
**Pull request by Great-Antique**
#14 Add the common trait which uses all Dont traits

### CODE REVIEW

File: src/Dont/DontUseMagic.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+trait DontUseMagic
+{
+    use DontGet;
+    use DontSet;
+    use DontCall;
+    use DontCallStatic;
+    use DontClone;
+    use DontSerialise;
+    use DontDeserialise;
+}
```
Comment (azjezz) [inline src/Dont/DontUseMagic.php]: new line

**Review by azjezz — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
Requires testing - possibly need to integrate this trait in the existing tests (with a data provider).

File: src/Dont/DontUseMagic.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dont;
+
+trait DontUseMagic
```
Comment (asgrim) [inline src/Dont/DontUseMagic.php]: `JustDont`

Comment (Great-Antique) [reply src/Dont/DontUseMagic.php]: @asgrim Are you serious?

Comment (Ocramius) [reply src/Dont/DontUseMagic.php]: I'd be fine with it as well :+1:

Comment (Great-Antique) [reply src/Dont/DontUseMagic.php]: Renamed

File: src/Dont/DontUseMagic.php (hunk 1 above)
Comment (Great-Antique) [reply src/Dont/DontUseMagic.php]: Fix

**Review by azjezz — APPROVED** [review]

**Review by asgrim — APPROVED** [review]
LGTM :+1: tests added, nicely done, thank you @Great-Antique :)

**Pull request merged**